### ISSUE TITLE
Cannot start Payara 5.Beta2 on Docker with jre8u161 (SupportedEllipticPointFormatsExtension), jre81u52 works

### ISSUE THREAD

**Issue by simdevmon** [payara/Payara#2344]
# Description #
----------

When I use the latest Oracle JRE 8 u161, I cannot start Payara 5.Beta2. It is working fine with the previous JRE 8 u152 release.

## Expected Outcome

I would expect that Payara works fine on the latest Oracle JRE 8 release.

## Current Outcome

There is no SSL connection available. In the logfile there is the following error:

```
[#|2018-01-31T10:49:37.796+0000|WARNING|Payara 5..Beta2|org.glassfish.grizzly.filterchain.DefaultFilterChain||
  GRIZZLY0013: Exception during FilterChain execution
java.lang.NoClassDefFoundError: sun/security/ssl/SupportedEllipticPointFormatsExtension
        at sun.security.ssl.HelloExtensions.<init>(HelloExtensions.java:84)
        at sun.security.ssl.HandshakeMessage$ClientHello.<init>(HandshakeMessage.java:362)
        at sun.security.ssl.ServerHandshaker.processMessage(ServerHandshaker.java:223)
        at sun.security.ssl.Handshaker.processLoop(Handshaker.java:984)
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:924)
        at sun.security.ssl.Handshaker$1.run(Handshaker.java:921)
        at java.security.AccessController.doPrivileged(Native Method)
        at sun.security.ssl.Handshaker$DelegatedTask.run(Handshaker.java:1379)
        at org.glassfish.grizzly.ssl.SSLUtils.executeDelegatedTask(SSLUtils.java:274)
        at org.glassfish.grizzly.ssl.SSLBaseFilter.doHandshakeStep(SSLBaseFilter.java:708)
        at org.glassfish.grizzly.ssl.SSLBaseFilter.doHandshakeStep(SSLBaseFilter.java:622)
        at org.glassfish.grizzly.ssl.SSLBaseFilter.handleRead(SSLBaseFilter.java:334)
        at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
        at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
        at org.glassfish.grizzly.portunif.PUFilter.handleRead(PUFilter.java:231)
        at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:119)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:284)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:201)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:133)
        at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:112)
        at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:77)
        at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:539)
        at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:112)
        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:117)
        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.access$100(WorkerThreadIOStrategy.java:56)
        at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:137)
        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:593)
        at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:573)
        at java.lang.Thread.run(Thread.java:748)
```

## Context

I did some investigation and I had a look at the **/jre/lib/jsse.jar** file of both **JRE 8u152** and **JRE 8u161** and saw that the class was removed in **u161**.

![2018-01-31_12h07_36](https://user-images.githubusercontent.com/16277781/35619772-57fe2614-067f-11e8-808a-0a8911e9db43.png)

## Environment ##

- **Payara Version**: 5.Beta2
- **Edition**: Full
- **JDK Version**: Oracle 8u161
- **Operating System**: Linux

**Comment by svendiedrichsen** [comment]
Your comparison also shows that you are trying to run Payara using a JRE now. Previously you have used a JDK which IMO is the way it should be.

**Comment by simdevmon** [comment]
@svendiedrichsen Yes, on Docker I run installed the latest Oracle JRE 8u162 where the error occured. 
Only the screenshots are from Windows and show the content of **jsse.jar**, which is part of the JRE.

**Comment by svendiedrichsen** [comment]
@simdevmon IMO you cannot run GF/Payara using a JRE. See GF doc here https://javaee.github.io/glassfish/doc/5.0/installation-guide.pdf - Installation requirements.
> ..If necessary, download and install the required JDK software...

**Comment by simdevmon** [comment]
@svendiedrichsen I had a typo in my previous post. I had it running on JDK, not JRE.

**Comment by smillidge** [comment]
looks like the removal of these classes is causing problems with Grizzly NPN

**Comment by smillidge** [comment]
Fixed by #2367

**Comment by smillidge** [comment]
Fixed by #2419
